cirq-core/cirq/study/result.py

```diff
@@ -266,50 +247,101 @@ def histogram(  # type: ignore
         """
         return self.multi_measurement_histogram(keys=[key], fold_func=lambda e: fold_func(e[0]))
 
-    def __repr__(self) -> str:
-        def item_repr(entry):
-            key, val = entry
-            return f'{key!r}: {proper_repr(val)}'
-
-        measurement_dict_repr = (
-            '{' + ', '.join([item_repr(e) for e in self.measurements.items()]) + '}'
-        )
-
-        return f'cirq.Result(params={self.params!r}, measurements={measurement_dict_repr})'
-
-    def _repr_pretty_(self, p: Any, cycle: bool) -> None:
-        """Output to show in ipython and Jupyter notebooks."""
-        if cycle:
-            # There should never be a cycle.  This is just in case.
-            p.text('Result(...)')
-        else:
-            p.text(str(self))
-
-    def __str__(self) -> str:
-        return _keyed_repeated_bitstrings(self.measurements)
-
-    def __eq__(self, other):
-        if not isinstance(other, type(self)):
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Result):
             return NotImplemented
         return self.data.equals(other.data) and self.params == other.params
 
-    def _measurement_shape(self):
+    def _measurement_shape(self) -> Tuple['cirq.ParamResolver', Dict[str, Tuple[int, ...]]]:
         return self.params, {k: v.shape[1] for k, v in self.measurements.items()}
 
     def __add__(self, other: 'cirq.Result') -> 'cirq.Result':
-        if not isinstance(other, type(self)):
+        if not isinstance(other, Result):
             return NotImplemented
         if self._measurement_shape() != other._measurement_shape():
             raise ValueError(
-                'TrialResults do not have the same parameters or do '
-                'not have the same measurement keys.'
+                'Results do not have the same parameters or do not have the '
+                'same measurement keys.'
             )
         all_measurements: Dict[str, np.ndarray] = {}
         for key in other.measurements:
             all_measurements[key] = np.append(
                 self.measurements[key], other.measurements[key], axis=0
             )
-        return Result(params=self.params, measurements=all_measurements)
+        return ResultImpl(params=self.params, measurements=all_measurements)
+
+
+class ResultImpl(Result):
```

Comment:
With an eye towards #3868, it may be preferable to name this implementation something that clearly indicates that its "default" storage type is keyed measurements (as opposed to raw values, pandas DataFrame, or something more exotic like IQ points).

Reply:
I'm certainly open to renaming this, but not sure what a good name would be. Any specific suggestions?

Reply:
The simplest would be "KeyedResultImpl" or "MeasurementKeyResultImpl" - the others could then be e.g. "RawResultImpl" and "DataFrameResultImpl".